src/wallet/spend.cpp

```diff
@@ -1392,4 +1392,381 @@ bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet,
+};
+
+void SpoofTransactionFingerprint(CMutableTransaction& tx, FastRandomContext& rng_fast)
+{
+    // Transaction "fingerprint" spoofing
+    struct Fingerprint {
+        bool standardVersion = false;
+        bool antiFeeSniping = false;
+        bool bip69Ordering = false;
+    };
+
+    // wallet fingerprints based on info from variuous sources, see:
+    // https://github.com/achow101/wallet-fingerprinting/blob/main/fingerprints.md
+    // https://gitlab.com/1440000bytes/goldfish
+    // https://ishaana.com/blog/wallet_fingerprinting/
+    static const Fingerprint s_walletFingerprints[] = {
+        {true, true, false},   // Core
+        {true, true, true},    // Electrum
+        {true, false, false},  // Blue
+        {false, false, true},  // Trezor
+        {false, false, false}, // Trust, Ledger
+    };
+    constexpr size_t NUM_WALLET_FINGERPRINTS = sizeof(s_walletFingerprints) / sizeof(s_walletFingerprints[0]);
+
+    auto fingerprintIndex = rng_fast.randrange(NUM_WALLET_FINGERPRINTS);
+    const Fingerprint& fingerprint = s_walletFingerprints[fingerprintIndex];
+
+    if (fingerprint.standardVersion) {
+        Assert(tx.nVersion == TX_MAX_STANDARD_VERSION);
+    } else {
+        tx.nVersion = 1;
+    }
+
+    if (fingerprint.antiFeeSniping) {
+        // By default "Core" implements anti-fee-sniping (nLockTime == block_height - rng_fast.randrange(100))
+    } else {
+        // no anti-fee-sniping
+        tx.nLockTime = 0;
+    }
+
+    if (fingerprint.bip69Ordering) {
+        // Sort the inputs and outputs in accordance with BIP69
+        auto sortInputsBip69 = [](const CTxIn& a, const CTxIn& b) {
+            // COutPoint operator< does sort in accordance with Bip69, so just use that.
+            return a.prevout < b.prevout;
+        };
+        std::sort(tx.vin.begin(), tx.vin.end(), sortInputsBip69);
+
+        auto sortOutputsBip69 = [](const CTxOut& a, const CTxOut& b) {
+            if (a.nValue == b.nValue) {
+                // Note: prevector operator< does NOT properly order scriptPubKeys lexicographically. So instead we
+                // fall-back to using std::memcmp.
+                const auto& spkA = a.scriptPubKey;
+                const auto& spkB = b.scriptPubKey;
+                const int cmp = std::memcmp(spkA.data(), spkB.data(), std::min(spkA.size(), spkB.size()));
+                return cmp < 0 || (cmp == 0 && spkA.size() < spkB.size());
+            }
+            return a.nValue < b.nValue;
+        };
+        std::sort(tx.vout.begin(), tx.vout.end(), sortOutputsBip69);
+    } else {
+        // By default "Core" doesn't perform BIP69 ordering
+    }
+}
+
+util::Result<CreatedTransactionResult> CreateDeniabilizationTransaction(CWallet& wallet, const std::set<COutPoint>& inputs, const std::optional<OutputType>& opt_output_type, unsigned int confirm_target, unsigned int deniabilization_cycles, bool sign, bool& insufficient_amount)
```

Comment:
same

Reply:
The "inputs", "out_output_type" and "confirm_target" are usually passed in as part of CCoinControl. 
However for deniabilization we create a specifically built CCoinControl, so I didn't want the API user to have to fill one out, just to have the majority of the fields essentially ignored, so that's why I decided to list these individually.
I could make a new class to pass them together though, eg CDeniabilizationControl? 
Does that sound reasonable?

Reply:
just by thinking of the possibility seems reasonable enough for me at this point

for now make as you feel it's best made